nuxeo-connect-client/src/main/java/org/nuxeo/connect/connector/AbstractConnectConnector.java

```diff
@@ -218,19 +221,23 @@ public DownloadingPackage getDownload(String id) throws ConnectServerError {
         // Fallback on the real source
         String url = getBaseUrl() + GET_DOWNLOADS_SUFFIX + "/" + urlSuffix;
         ConnectServerResponse response = execCall(url);
-        String json = response.getString();
-        response.release();
         try {
+            String json = response.getString();
+            if (json == null) { // Not found
+                return result;
```

Comment:
It's confusing to have this `return result` here and another one at a different level outside the `try`. Either put the whole logic in a `if (json != null)` and just keep the last `return result`, or move the last `return result` before the `catch`.